### ISSUE TITLE
Go-to-definition only works for bower projects in Atom - due to sourceGlobs / packagePath config mismatch

### ISSUE THREAD

**Issue by Dretch** [nwolverson/purescript-language-server#47]
Right now go-to-definition does not work for Spago projects (even when setting the source paths to include the `.spago` folder), and I think this is why:
 - atom-ide-purescript sends the source paths globs config in the `sourceGlobs` field
 - purescript-language-server ignores `sourceGlobs` completely, and looks in the `packagePath` field to get the source globs. The default value for `packagePath` is `"bower_components"`, so bower projects work okay.
- vscode-ide-purescript apparently uses `packagePath`, so I guess VSCode works with custom source paths (I have not tried) - https://github.com/nwolverson/vscode-ide-purescript/blob/d820e73d5b4564feb2111aafe1daa0341f331b7e/package.json#L113

I changed purescript-language-server locally so it looked at `sourceGlobs`, and go-to-definition starting working with my `.spago` folder... but making the change would likely break vscode, which is not good. I need to do some more digging before I can suggest a proper fix.

**Comment by nwolverson** [comment]
I think this https://github.com/nwolverson/purescript-language-server/issues/43 would solve your issue? Assuming the config is propagated.

Other than that we can either add support for `sourceGlobs` or, since it's currently broken already for atom, switch it to the vscode plugin way of doing it.

**Comment by Dretch** [comment]
> I think this #43 would solve your issue? Assuming the config is propagated.

Yup, I think #43 would be the best solution for supporting Spago.

> Other than that we can either add support for sourceGlobs or, since it's currently broken already for atom, switch it to the vscode plugin way of doing it.

I missed something when I looked at this before. The vscode plugin way only allows passing a path to single folder that is expected to be laid out like `bower_components`, whereas the atom way allows (if it worked), passing an array of arbitrary globs, which is much more flexible.
